core/src/main/scala/org/bykn/bosatsu/Evaluation.scala

```diff
@@ -349,7 +364,7 @@ case class Evaluation(pm: PackageMap.Inferred, externals: Externals) {
                         splice match {
                           case None => (acc1, next)
                           case Some(nm) =>
-                            val rest = Eval.later(VList(spliceVals.reverse))
+                            val rest = Eval.now(VList(spliceVals.reverse))
```

Comment:
Is this related to structural recursion support. Was it an attempt at improving performance?

Reply:
part of the hacks to improve performance (still not great, but I did get the tests to pass).